src/DaedalusCompiler/Compilation/DaedalusParserListener.cs

```diff
@@ -381,7 +400,14 @@ public override void ExitComplexReference(DaedalusParser.ComplexReferenceContext
         {
             var referenceNodes = context.complexReferenceNode();
             var symbolPart = referenceNodes[0];
-            var arrIndex = symbolPart.simpleValue();
+
+            int arrIndex = 0;
+            var arrIndexContext = symbolPart.simpleValue();
+            if (arrIndexContext != null)
+            {
+                arrIndex = int.Parse(arrIndexContext.GetText());
```

Comment:
You can't assume that `arrIndexContext ` is number, it could be Identifier as well, like here: ```s = some.field[SOME_VAR]```

Reply:
Done

Reply:
PS. actually it cannot be SOME_VAR, but it can be SOME_CONST